### PR TITLE
Implement release-validate-deps with kotlin

### PR DESCRIPTION
**Pull request by flyingsilverfin**
We implement release validation with kotlin, adding new macro and rule to generate a templated kotlin script.

### CODE REVIEW

File: tool/release/rules.bzl [hunk 1]
```diff
@@ -16,42 +16,29 @@
...
-            executable = test_script,
-            runfiles = ctx.runfiles(files = [test_script, ctx.file.refs])
+            runfiles = ctx.runfiles(files = [ctx.file.refs]),
+            files = depset(direct = [test_script])
         )
     ]
 
 
-release_validate_deps_test = rule(
+release_validate_deps_script = rule(
```
Comment (vmax) [inline tool/release/rules.bzl:41]: @flyingsilverfin can we prefix it with `_` so it's clear that this rule is not made for public consumption?

Comment (flyingsilverfin) [reply tool/release/rules.bzl:41]: yep ok

File: tool/release/BUILD [hunk 2]
```diff
@@ -27,8 +30,11 @@ py_binary(
...
-    main = "validate-deps.py",
-)
+kt_jvm_binary(
+    name = "validate-deps-refs",
+    main_class = "tool.release.Validate_depsKt",
+    srcs = ["ValidateDeps.kt"],
+    deps = [
+        "@maven//:com_eclipsesource_minimal_json_minimal_json"
+    ]
+)
```
Comment (vmax) [inline tool/release/BUILD:40]: @flyingsilverfin newline please

Comment (flyingsilverfin) [reply tool/release/BUILD]: done, and added license

**Pull request merged**